extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java

```diff
@@ -317,6 +317,10 @@
             result.add(new DecoratorBuildItem(target, new ApplyRequestsMemoryDecorator(name, m)));
         });
 
+        if (config.getSecurityContext().isAnyPropertySet()) {
```

Comment:
This is a hack ... But I tried to make the security context optional (which it is, so it makes sense), but the configuration does not deal well with having a complex and nested configuration with optional. Concretely, if I use Optional instead of the hack `isAnyPropertySet`, it fails with:

```
java.lang.RuntimeException: java.lang.RuntimeException: java.lang.IllegalArgumentException: Can not set io.quarkus.kubernetes.deployment.SecurityContextConfig$WindowsOptions field io.quarkus.kubernetes.deployment.SecurityContextConfig.windowsOptions to java.util.Optional
```

This is related to https://github.com/quarkusio/quarkus/issues/7862, which was recently closed with a workaround, but not to really give support of optional configuration groups.

Reply:
From my point of view this is clear as it communicates the intention and there is no room for interpretations.
Going the Optional way it would be unclear how default values are meant to be handled. 
So, +1 for this.